browser/components/CodeEditor.js

```diff
@@ -274,7 +354,11 @@ export default class CodeEditor extends React.Component {
       extraKeys: this.defaultKeyMap
     })
 
-    this.setMode(this.props.mode)
+    if (!this.props.mode && this.props.value) {
```

Comment:
Should you check for `config.editor.snippetDefaultLanguage === 'Auto Detect'` here too?

Reply:
Updated